src/relay/source_map.cc

```diff
@@ -0,0 +1,70 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file source_map.cc
+ * \brief Source maps for Relay.
+ */
+
+#include <tvm/relay/logging.h>
+#include <tvm/relay/source_map.h>
+#include <iostream>
+
+namespace tvm {
+namespace relay {
+
+using tvm::IRPrinter;
+using namespace tvm::runtime;
+
+SourceFragment::SourceFragment(std::string file_name, std::string source)
+    : file_name(file_name), source_lines({}) {
+  RELAY_LOG(INFO)<< "SourceFragment::SourceFragment source=" << source << std::endl;
+  std::stringstream source_stream;
+  source_stream.str(source.c_str());
+  std::string line;
+
+  while (std::getline(source_stream, line)) {
+    RELAY_LOG(INFO) << "SourceFragment::SourceFragment: line=" << line << std::endl;
+    std::string copy(line);
+    source_lines.push_back(copy);
+  }
+}
+
+std::string SourceFragment::SourceAt(Span sp, int max_lines) {
```

Comment:
Does it make sense to default max_lines to 0, and return the current line in that case.

Reply:
What about defaulting to `1`? unless I'm missing something?

Reply:
Sorry, I probably misunderstood it. 1 works if it means the current line.